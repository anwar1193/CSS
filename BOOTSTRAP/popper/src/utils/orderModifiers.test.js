// @flow
import orderModifiers from './orderModifiers';

const a = {
  name: 'applyStyles',
  phase: 'write',
};
const b = {
  name: 'arrow',
  phase: 'main',
  requiresIfExists: ['preventOverflow'],
};
const c = {
  name: 'computeStyles',
  phase: 'afterMain',
};
const d = {
  name: 'detectOverflow',
  phase: 'read',
  requires: ['popperOffsets'],
};
const e = {
  name: 'eventListeners',
  phase: 'write',
};
const f = {
  name: 'flip',
  phase: 'main',
  requires: ['detectOverflow'],
  requiresIfExists: ['offset'],
};
const g = {
  name: 'hide',
  phase: 'main',
  requires: ['detectOverflow'],
};
const h = {
  name: 'offset',
  phase: 'main',
  requires: ['popperOffsets', 'detectOverflow'],
};
const i = {
  name: 'popperOffsets',
  phase: 'read',
};
const j = {
  name: 'preventOverflow',
  phase: 'main',
  requires: ['detectOverflow'],
  requiresIfExists: ['offset'],
};

const modifiers: any = [a, b, c, d, e, f, g, h, i, j];

it('should order modifiers based on `phase` and `requires`', () => {
  expect(orderModifiers(modifiers)).toMatchSnapshot();
});
