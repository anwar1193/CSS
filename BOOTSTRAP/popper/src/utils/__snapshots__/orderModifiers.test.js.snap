// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should order modifiers based on \`phase\` and \`requires\` 1`] = `
Array [
  Object {
    "name": "popperOffsets",
    "phase": "read",
  },
  Object {
    "name": "detectOverflow",
    "phase": "read",
    "requires": Array [
      "popperOffsets",
    ],
  },
  Object {
    "name": "offset",
    "phase": "main",
    "requires": Array [
      "popperOffsets",
      "detectOverflow",
    ],
  },
  Object {
    "name": "preventOverflow",
    "phase": "main",
    "requires": Array [
      "detectOverflow",
    ],
    "requiresIfExists": Array [
      "offset",
    ],
  },
  Object {
    "name": "arrow",
    "phase": "main",
    "requiresIfExists": Array [
      "preventOverflow",
    ],
  },
  Object {
    "name": "flip",
    "phase": "main",
    "requires": Array [
      "detectOverflow",
    ],
    "requiresIfExists": Array [
      "offset",
    ],
  },
  Object {
    "name": "hide",
    "phase": "main",
    "requires": Array [
      "detectOverflow",
    ],
  },
  Object {
    "name": "computeStyles",
    "phase": "afterMain",
  },
  Object {
    "name": "applyStyles",
    "phase": "write",
  },
  Object {
    "name": "eventListeners",
    "phase": "write",
  },
]
`;
